src/App.tsx

```diff
@@ -2,7 +2,7 @@ import React from "react";
 import { IonApp } from "@ionic/react";
 import { IonReactRouter } from "@ionic/react-router";
 import MainRouter from "components/MainRouter/MainRouter";
-import AuthProvider from "components/AuthProvider";
+import AuthProvider from "components/AuthProvider/AuthProvider";
```

Comment:
You can create "components/AuthProvider/index.tsx" instead of "components/AuthProvider/AuthProvider.tsx"

And importing can be done as "import AuthProvider from "components/AuthProvider"

Reply:
@bharat619 I actually considered doing that early on, however, i noticed that it actually makes fuzzy search harder and in the IDE it becomes more difficult to differentiate tabs. So I decided with this convention.

For more info refer to: https://bradfrost.com/blog/post/this-or-that-component-names-index-js-or-component-js/

Reply:
I do think having `name/name.js` is "better" in many ways, but I always fall back to `name/index.js` because the redundancy bothers me to death 😄 

Note, if you move the specs to their own folder, you can have `components/AuthProvider.tsx` without cluttering the folders.

There's no right or wrong way here, I'm okay with any of the above